app/db/impl/model/common/StateClassifiable.scala

```diff
@@ -0,0 +1,22 @@
+package db.impl.model.common
+
+import db.Model
+import db.impl.table.common.StateColumn
+import models.project.ProjectStates.ProjectState
+
+/**
+  * Represents a [[Model]] that has a declared state.
+  */
+trait StateClassifiable extends Model { self =>
+
+  override type M <: StateClassifiable { type M = self.M }
+  override type T <: StateColumn[M]
+
+  /**
+    * Returns the current state of the [[Model]].
+    *
+    * @return the current state
+    */
+  def state: ProjectState
```

Comment:
This is not really a common trait when the type is ProjectState but I don't know what to do here.

Reply:
I make quite a bit of use of this trait in the version visibility (or now state PR)

Reply:
Ok, so I leave it like this for now and you change it then in your PR?